util/db/dbutil_test.go

```diff
@@ -333,3 +335,41 @@ func TestDBConcurrencyRW(t *testing.T) {
 	}
 
 }
+
+type WarningLogCounter struct {
+	logging.Logger
+	warningsCounter int
+}
+
+func (wlc *WarningLogCounter) Warnf(string, ...interface{}) {
+	wlc.warningsCounter++
+}
+
+func (wlc *WarningLogCounter) With(key string, value interface{}) logging.Logger {
+	return wlc
+}
+
+// Test resetting warning notification
+func TestResettingTransactionWarnDeadline(t *testing.T) {
+	acc, err := MakeAccessor("fn.db", false, true)
+	require.NoError(t, err)
+	defer acc.Close()
+	logger := WarningLogCounter{
+		Logger: logging.Base(),
+	}
+	acc.log = &logger
+	err = acc.Atomic(func(ctx context.Context, tx *sql.Tx) error {
+		time.Sleep(1001 * time.Millisecond)
```

Comment:
I'd prefer seeing timeouts configurable, just not to waste test running time. but we can leave with 2 seconds if you add `t.Parallel()`

Reply:
I'd rather avoid having it configurable in this change, but agree that we should add that in the future.
I have updated this one to use Parallel so that we can run both of the tests at the same time.